modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -286,41 +377,51 @@ function AddFDSFilterModalContent({
 					/>
 				</ClayForm.Group>
 
-				<ClayForm.Group>
+				<ClayForm.Group
+					className={classNames({
+						'has-error': fieldInUseValidationError,
+					})}
+				>
 					<label htmlFor={selectedFieldFormElementId}>
 						{Liferay.Language.get('filter-by')}
 					</label>
-
-					<ClaySelectWithOption
-						aria-label={Liferay.Language.get('filter-by')}
-						disabled={!!filter}
-						onChange={(event) => {
-							const newVal = fields.find(
-								(item) => item.name === event.target.value
-							);
+					<CellRendererDropdown
+						cellRenderers={fields}
+						inUseFields={inUseFields}
+						namespace={namespace}
+						onItemClick={(item: IField) => {
+							const newVal = fields.find((field) => {
+								return field.name === item.label;
+							});
 
 							if (newVal) {
+								if( inUseFields.includes(newVal.name) ) {
+									setFieldInUseValidationError(true)
+									setSaveButtonDisabled(true)
+								} else {
+									setFieldInUseValidationError(false)
+									setSaveButtonDisabled(false)
+								}
```

Comment:
This might be a bit cleaner without the if/else
```js
const inUse = inUseFields.includes(newVal.name);

setFieldInUseValidationError(inUse)
setSaveButtonDisabled(inUse)
```

Reply:
Absolutely, done